File: ocaml/examples/hidden.ml
open Help_tree

let () =
  let opts = discovery_options () in

  let hidden_sub = {
    name = "debug";
    description = "Hidden debug command";
    hidden = true;
    options = [
      { name = "dump"; short = Some 'd'; description = "Dump internal state"; hidden = true; arg = None };
    ];
    arguments = [];
    subcommands = [];
  } in

  let visible_sub = {
    name = "run";
    description = "Run the application";
    hidden = false;
    options = [
      { name = "mode"; short = Some 'm'; description = "Run mode"; hidden = false; arg = Some "MODE" };
    ];
    arguments = [
      { name = "target"; description = "Target to run"; required = true; hidden = false };
    ];
    subcommands = [];
  } in

  let root = {
    name = "hidden";
    description = "An example with hidden commands and options";
    hidden = false;
    options = [
      { name = "verbose"; short = Some 'v'; description = "Verbose output"; hidden = false; arg = None };
      { name = "secret"; short = Some 's'; description = "Hidden secret flag"; hidden = true; arg = Some "TOKEN" };
    ];
    arguments = [
      { name = "input"; description = "Input file"; required = false; hidden = true };
    ];
    subcommands = [visible_sub; hidden_sub];
  } in

  if should_render_tree opts then
    print_string (render opts root)
  else
    Printf.printf "Run with --help-tree to see the command tree.\n"

File: nim/examples/hidden.nim
import std/os
import help_tree

proc hiddenCmd(): TreeCommand =
  new(result)
  result.name = "hidden"
  result.description = "An example with hidden commands and flags"
  result.options = discoveryOptions() & @[
    TreeOption(name: "verbose", long: "--verbose", description: "Verbose output", required: false, takesValue: false),
    TreeOption(name: "debug", long: "--debug", description: "Enable debug mode", required: false, takesValue: false, hidden: true)
  ]

  var listCmd = TreeCommand(
    name: "list",
    description: "List items"
  )

  var showCmd = TreeCommand(
    name: "show",
    description: "Show item details",
    arguments: @[TreeArgument(name: "ID", description: "Item ID", required: true)]
  )

  var admin = TreeCommand(
    name: "admin",
    description: "Administrative commands",
    hidden: true
  )
  admin.subcommands.add(TreeCommand(name: "users", description: "List all users"))
  admin.subcommands.add(TreeCommand(name: "stats", description: "Show system stats"))
  admin.subcommands.add(TreeCommand(name: "secret", description: "Secret backdoor"))

  result.subcommands = @[listCmd, showCmd, admin]

when isMainModule:
  let root = hiddenCmd()
  var invocation = parseHelpTreeInvocation(commandLineParams())
  if invocation.helpTree:
    let configPath = joinPath(currentSourcePath().parentDir, "help-tree.json")
    if fileExists(configPath):
      let config = loadConfig(configPath)
      applyConfig(invocation.opts, config)
    runForParser(root, invocation.opts, invocation.path)
    quit(0)

  echo "Run with --help-tree to see the command tree."
